spec/kolekti/memory_persistence_strategy_spec.rb

```diff
@@ -18,10 +18,20 @@
       }
     }
 
-    it 'is expected to store the result in memory' do
-      subject.create_tree_metric_result(metric_configuration, module_name, value, granularity)
+    context 'with valid metric configuration' do
+      it 'is expected to store the result in memory' do
+        subject.create_tree_metric_result(metric_configuration, module_name, value, granularity)
+        expect(subject.tree_metric_results).to eq([result_hash])
+      end
+    end
 
-      expect(subject.tree_metric_results).to eq([result_hash])
+    context 'with duplicated module name or configuration' do
```

Comment:
By https://github.com/mezuro/kolekti/issues/19 this description and implementation should consider the `and` case instead of the  `or`.